### ISSUE TITLE
Monitoring API should expose pipeline state/update overall status to indicate there is an issue

### ISSUE THREAD

**Issue by robbavey** [elastic/logstash#13060]
Since #12307, it has been possible for pipelines to shut down, but the rest of logstash to remain running. 
This leads to logstash running in an 'impaired' state where only a subset of pipelines are up, and without trawling the logs for pipeline shutdown events, no obvious way to understand that logstash is in such a state. This has particular impact until #12688 is solved, as the stopped/crashed pipelines will not restart without active intervention.

The monitoring API should reflect the fact that Logstash is in this state by exposing this information in (at least) a couple of different ways:

- Update the overall `status`, so it is no longer `GREEN`
- Add pipeline status to each of the pipelines

**Comment by JeffSaxeVA** [comment]
Agreed, this seems a useful enhancement. I'm just now trying an output plug-in that happens to send to an HTTPS destination on the Internet, and if _anything is disrupted_ (timeout or network interruption, untrusted SSL cert, their server refused to accept my POST body, etc.), then the Ruby Net::HTTP code coughs out an exception, which bubbles all the way up to Logstash, which logs a single (and actually very helpful!) message to its own log, but then it shuts down that pipeline and leaves it down until I, the sysadmin, figure this out and go take an action to restart it. It _happens_ that my input here is a TCP listener, so I can fabricate my own off-box monitor for this condition by socket-testing the TCP port and alerting when it's down, but it would be nice to simply query the :9600 _node/pipelines (or _node/stats/pipelines) and see the JSON change in both the overall status and an individual status for each pipeline, as Rob suggests.

I have worked around this by modifying the plug-in's Ruby code, putting a "rescue" section around the potentially exception-producing line of code and then logging the warning, but throwing away the exception. Thus Logstash _doesn't actually stop_ the pipeline, but this obviously a hack.

By the way, let me say I'm very impressed with the whole monitoring API! A lot of software isn't even this self-aware, or at least doesn't have a way to query these useful stats in such a nice format. Especially helpful is the _node/stats/pipelines, showing how the CPU is being consumed in each of my string-parsing filter sections; and I haven't even tried the "hot threads" API, since I only recently started using the 7.x version of Logstash. Thank you for having this API at all!
